packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1606,17 +1605,6 @@ export class SnapController extends BaseController<
           );
         }
 
-        const permissions = this.messagingSystem.call(
-          'PermissionController:getPermissions',
-          origin,
-        ) as SubjectPermissions<PermissionConstraint>;
-
-        if (!isSnapPermitted(permissions, snapId)) {
```

Comment:
Should we leave this in, just in case we somehow trigger an install without the permission?

Reply:
We install the snap before the permission is granted so we cannot leave it here.

Reply:
To avoid having to rollback granted permissions if a side-effect fails we run the side effects in between the user approval and the permission grant so at the time the snap is installed the permission is technically not granted.

Reply:
But the changes made here implies that you call `requestPermissions` to install a snap so we can safely assume that the permission is granted if a snap is installed